pkg/internal/sdk/template/brc29/brc29_opts.go

```diff
@@ -0,0 +1,32 @@
+package brc29
+
+import sighash "github.com/bsv-blockchain/go-sdk/transaction/sighash"
+
+type lockOptions struct {
+	mainNet bool
+}
```

Comment:
```suggestion
// Network represents the blockchain network.
type Network int

const (
	MainNet Network = iota
	TestNet
)

type lockOptions struct {
	network Network
}
```

WDYT ? for me looks nicer :)

Reply:
I started with that,
But then I will need to convert it back to boolean. 
This is implementation detail that is driven by the needs of script.Address function. No one from outside should bother about it.